app/src/androidTest/java/org/simple/clinic/AuthenticationRule.kt

```diff
@@ -46,28 +73,61 @@ class AuthenticationRule : TestRule {
         try {
           // Login also needs to happen inside this try block so that in case
           // of a failure, logout() still gets called to reset all app data.
-          register()
+          registeredFacilityUuid = register()
+          if(registerPatientWithUuid != null) {
+            registerPatient(patientUuid = registerPatientWithUuid)
+          }
+          if(registerAppointmentWithUuid != null) {
+            registerAppointment(appointmentUuid = registerAppointmentWithUuid)
+          }
           base.evaluate()
 
         } finally {
           logout()
+          registeredFacilityUuid = null
         }
       }
     }
   }
 
-  private fun register() {
+  private fun registerPatient(patientUuid: UUID) {
```

Comment:
Discussed offline: We'll move registration of patients and appointments to somewhere outside this class because:

- they're not the responsibility of `AuthenticationRule`
- the user's current facility can be fetched easily using `FacilityRepository.currentFacility(User)`